src/edu/csus/ecs/pc2/ui/ResultsComparePane.java

```diff
@@ -0,0 +1,461 @@
+             for (String name : filesCreated) {
+                messageLines.add(name);
+            }
+             
+             String message = String.join(Constants.NL, messageLines);
+             showMessage(this, "Files created", message);
+             
+             return true;
+             
+        } catch (Exception e) {
+            // TODO REFACTOR present error message and stacktrace to user with to be written standard error dialog
+            getController().getLog().log(Level.WARNING,"Unable to create export files to "+pc2ResultsDirectory+" "+e.getMessage(), e);
+            showErrorMessage("Unable to create export files to "+pc2ResultsDirectory+" "+e.getMessage(), true);
+        }
+        
+        return false;
+    }
+
+    protected void viewCompareResultsFiles() {
+
+        String pc2ResultsDirectory = pc2ResultsDirectoryTextField.getText();
+
+        String primaryCCSDirectory = primaryCCSResultsDirectoryTextField.getText();
+
+        if (showErrorMessage("Enter a primary CCS results directory", StringUtilities.isEmpty(primaryCCSDirectory))) {
+            return;
+        }
+
+        if (showErrorMessage("Primary CCS directory does not exist, pick an existing results directory", !directoryExists(primaryCCSDirectory))) {
+            return;
+        }
+
+        if (showErrorMessage("Enter a pc2 results directory", StringUtilities.isEmpty(pc2ResultsDirectory))) {
+            return;
+        }
+        
+        if (showErrorMessage("Missing pc2 directory - use Export Results to create results files", !directoryExists(pc2ResultsDirectory))) {
+            return;
+        }
+
+        try {
+            
+            ResultsCompareReport report = new ResultsCompareReport(getContest(), getController(), primaryCCSDirectory, pc2ResultsDirectory, showDetailsCheckbox.isSelected(), MISSING_SOURCE_TITLE, MISSING_TARGET_TITLE);
+            Utilities.viewReport(report, report.getPluginTitle(), getContest(), getController(), true);
+        } catch (Exception e) {
+            getLog().log(Level.WARNING, "Exception trying export and compare results" + e.getMessage(), e);
+            showMessage(this, "Problem comparing results", "Error writing or comparing results " + e.getMessage());
+        }
+    }
+
+
+    private boolean directoryExists(String dirname) {
+        // TODO REFACTOR move this to a utility class
+        if (StringUtilities.isEmpty(dirname)){
+            return false;
+        }
+        
+        return new File(dirname).isDirectory();
+    }
+
+    /**
+     * Show message if showMessage is true.
+     * 
+     * @param message
+     *            describe what to do
+     * @param showMessage
+     *            should the mesage be shown
+     * 
+     * @return the input condition specified by showMessage
+     */
+    private boolean showErrorMessage(String message, boolean showMessage) {
+        if (showMessage) {
+            showMessage(this, "Correct error, then try again", message);
```

Comment:
I would suggest adding a colon and space after the word "again".  Otherwise, the phrase "Correct error, then try again" runs right into the message being displayed.

Reply:
I do not agree, the first string is the dialog title, I doubt that runs right into the message.